### PR TITLE
Add a template

### PR DESCRIPTION
**Pull request by snoyberg**
This is intended for usage with `stack new`. The code here is certainly up for improvement.

### CODE REVIEW

File: etc/template/LICENSE [hunk 1]
```diff
@@ -0,0 +1,30 @@
+Copyright Author name here (c) 2018
```
Comment (mgsloan) [inline etc/template/LICENSE]: Should probably be `AUTHOR` not `Author name here`

File: etc/template/package.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+# category:            Web
+
+# To avoid duplicated efforts in documentation and dealing with the
+# complications of embedding Haddock markup inside cabal files, it is
+# common to point users to the README.md file.
+description:         DESCRIPTION
+
+dependencies:
+- base >= 4.10 && < 5
+- rio >= 0.1.1.0 && < 0.2
```
Comment (roman) [inline etc/template/package.yaml]: Wondering why the upper-bound here?

Comment (snoyberg) [reply etc/template/package.yaml]: Good point, I've removed it.

File: etc/make-template.hs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+  -- Make sure it passes
+  proc "stack" ["test"] runProcess_
+
+  rawout <- withWorkingDir templatedir
+          $ proc "git" ["ls-files", "-z"]
+            readProcessStdout_
+  files <- mapM decode $ splitNulls $ BL.toStrict rawout
+  let src = forM_ files $ \fp -> yield (fp, readFileBinary $ templatedir </> fp)
+  runConduitRes $ src .| createTemplate .| sinkFile "rio.hsfiles"
```
Comment (roman) [inline etc/make-template.hs]: Oh 🙀🙀🙀 this is awesome! No longer nasty huge `hsfiles`, I'm copying this to my dotfiles like yesterday! No longer generating `Makefile`s with the wrong spacing 🎉

**Pull request merged**